generator/src/main/java/com/scottlogic/deg/generator/restrictions/TypeRestrictionsMerger.java

```diff
@@ -12,7 +12,7 @@
         final TypeRestrictions merged = left.intersect(right);
 
         if (merged == null) {
-            return new MergeResult();
+            return new MergeResult<>(DataTypeRestrictions.NO_TYPES_PERMITTED);
```

Comment:
If there is no intersection of allowed types, then no types are permitted.
This means that `null` can still be emitted as it is not impacted by the types restriction.